docker/docker_image_dest.go

```diff
@@ -394,11 +394,13 @@ func (d *dockerImageDestination) TryReusingBlobWithOptions(ctx context.Context,
 			continue
 		}
 
-		return true, private.ReusedBlob{
-			Digest:               candidate.Digest,
-			Size:                 size,
-			CompressionOperation: compressionOperation,
-			CompressionAlgorithm: compressionAlgorithm}, nil
+		if options.RequiredCompression == nil || (options.RequiredCompression == compressionAlgorithm) {
```

Comment:
Hum, sadly `compression.AlgorithmByName` returns a `compression.Algorithm` value, so the `*compression.Algorithm` pointer is not a unique identifier that can be compared for equality; I think this needs to compare `.Name()`.

Reply:
Done.